src/providers/ipa/ipa_s2n_exop.c

```diff
@@ -56,6 +58,55 @@ enum response_types {
     RESP_NAME_LIST
 };
 
+static const char *extdom_oids[] = {
+    EXOP_SID2NAME_OID,
+    EXOP_SID2NAME_V1_OID,
+    EXOP_SID2NAME_V2_OID,
+    NULL
+};
+
+static const char* extdom_protocol_to_oid(enum extdom_protocol protocol)
+{
+    if(protocol >= EXTDOM_V0 && protocol <= EXTDOM_V2) {
+        return extdom_oids[protocol];
+    }
+
+    return NULL;
+}
+
+static enum extdom_protocol extdom_oid_to_protocol(const char *oid)
+{
+    enum extdom_protocol i;
+
+    if (oid == NULL) {
+        return EXTDOM_INVALID_VERSION;
+    }
+
+    for (i = EXTDOM_V0; extdom_oids[i] != NULL; ++i) {
+        if (strcmp(extdom_oids[i], EXOP_SID2NAME_OID) == 0) {
```

Comment:
Hi,

I guess you mean `if (strcmp(extdom_oids[i], oid) == 0) {`?

I would also suggest to not use `EXTDOM_V0` as start for the loop but just `0`. `EXTDOM_V0` is just an enum and I think it is better to not assume a fixed numerical value here.

bye,
Sumit